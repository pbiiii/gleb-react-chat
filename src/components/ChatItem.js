import React from 'react';
import PropTypes from 'prop-types';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Avatar from '@material-ui/core/Avatar';
import { Link } from 'react-router-dom';
import moment from 'moment';
import ChatType from '../types/ChatType';

const ChatItem = ({ chat, active, disabled }) => {
    const { _id: chatId, title, createdAt } = chat;
    const avatarTitle = title.substring(0, 2).toUpperCase();
    return (
        <ListItem
            button
            component={Link}
            to={`/chat/${chatId}`}
            selected={active}
            disabled={disabled}
        >
            <Avatar>{avatarTitle}</Avatar>
            <ListItemText primary={title} secondary={moment(createdAt).fromNow()} />
        </ListItem>
    );
};

ChatItem.propTypes = {
    chat: PropTypes.shape(ChatType).isRequired,
    active: PropTypes.bool.isRequired,
    disabled: PropTypes.bool.isRequired,
};

export default ChatItem;
