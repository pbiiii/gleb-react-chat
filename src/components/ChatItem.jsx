import React from 'react';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Avatar from '@material-ui/core/Avatar';
import { Link } from 'react-router-dom';
import moment from 'moment';

const ChatItem = ({ chat, active }) => {
    const {
        _id: chatId,
        title,
        createdAt,
    } = chat;
    const avatarTitle = title.substring(0, 2).toUpperCase();
    return (
        <ListItem
            button
            component={Link}
            to={`/chat/${chatId}`}
            selected={active}
        >
            <Avatar>{avatarTitle}</Avatar>
            <ListItemText primary={title} secondary={moment(createdAt).fromNow()} />
        </ListItem>
    );
};

export default ChatItem;
