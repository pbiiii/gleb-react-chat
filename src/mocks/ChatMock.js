import UserMock from './UserMock';

export default {
    _id: 'ashasdaksdmkl1mkl',
    creator: { ...UserMock, _id: 2 },
    title: 'Mock title',
};
