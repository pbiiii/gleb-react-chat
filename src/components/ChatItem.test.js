/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatItem from './ChatItem';
import ChatMock from '../mocks/ChatMock';

jest.mock('react-router-dom', () => () => 'Link');

const mockProps = {
    active: true,
    disabled: true,
    chat: { ...ChatMock },
}

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatItem {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('render correctly', () => {
    const tree = renderer
        .create(<ChatItem {...mockProps} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
