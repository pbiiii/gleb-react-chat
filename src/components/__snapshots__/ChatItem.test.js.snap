// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  className="MuiButtonBase-root-72 MuiButtonBase-disabled-73 MuiListItem-root-60 MuiListItem-default-63 MuiListItem-gutters-68 MuiListItem-disabled-66 MuiListItem-button-69 MuiListItem-selected-71"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="-1"
  to="/chat/ashasdaksdmkl1mkl"
>
  <div
    className="MuiAvatar-root-75 MuiAvatar-colorDefault-76"
  >
    MO
  </div>
  <div
    className="MuiListItemText-root-78"
  >
    <span
      className="MuiTypography-root-84 MuiTypography-subheading-91 MuiListItemText-primary-81"
    >
      Mock title
    </span>
    <p
      className="MuiTypography-root-84 MuiTypography-body1-93 MuiTypography-colorTextSecondary-117 MuiListItemText-secondary-82"
    >
      a few seconds ago
    </p>
  </div>
</div>
`;
